### PR TITLE
Add Aesara backend

### PR DESCRIPTION
**Pull request by rlouf**
## Description

This PR adds an Aesara backend to `opt_einsum` following the discussion in #213.

## Todos
- Add Aesara backend
- Add relevant tests
- Updated dev environment
- Updated documentation and README

## Status
- [ ] Ready to go

### CODE REVIEW

File: opt_einsum/tests/test_backends.py [hunk 1]
```diff
@@ -212,6 +222,73 @@ def test_theano_with_sharing(string):
...
+    # check can call with numpy still
+    res_got2 = expr(var, backend="numpy")
+    assert np.allclose(res_exp, res_got2)
+
+    # check aesara call returns aesara still
+    res_got3 = expr(backends.to_aesara(var))
+    assert isinstance(res_got3, aesara.tensor.TensorVariable)
+
+
+@pytest.mark.skipif(not found_aesara, reason="Aesara not installed.")
```
Comment (dgasmith) [inline opt_einsum/tests/test_backends.py:269]: We really need to turn these into a fixture at some point so we're not duplicating data.

File: opt_einsum/tests/test_paths.py [hunk 2]
```diff
@@ -135,7 +132,6 @@ def test_explicit_path():
 
 
 def test_path_optimal():
-
```
Comment (dgasmith) [inline opt_einsum/tests/test_paths.py:138]: Do these new lines come from a black version bump?

**Review by dgasmith — APPROVED** [review]
Nice, this LGTM!

@jcmgray Do you want to get eyes on the PR as well before merging?

File: opt_einsum/tests/test_paths.py (hunk 2 above)
Comment (jcmgray) [reply opt_einsum/tests/test_paths.py:138]: I believe so, noticed this in a few other libraries as well.

**Comment by jcmgray** [comment]
All looks good to me, looking forward to testing the performance some time! 

By the way @rlouf, I have an implementation of pairwise `einsum` that should be compatible with with `aesara` here - https://github.com/jcmgray/einsum_bmm, if `aesara` needed an implementation. (@dgasmith its also faster than `numpy.einsum` in many cases, might be something worth including here too....)

**Comment by rlouf** [comment]
That looks awesome! We have an issue that's been open for a while to implement `einsum`, it would be great to have it in the library.

**Comment by dgasmith** [comment]
@rlouf Looks like Asera tests are failing.

**Comment by dgasmith** [comment]
@rlouf Gentle ping here.

**Comment by dgasmith** [comment]
@rlouf Another gentle ping here, we will be shipping another release within the next week.

**Comment by rlouf** [comment]
I am sorry but I am not working on Aesara anymore. I am closing the PR for now.

**Pull request closed without merging**